src/execution/mod.rs

```diff
@@ -288,20 +357,30 @@ where
             let (module_idx, func_idx) =
                 self.get_indicies(&function_ref.module_name, &function_ref.function_name)?;
 
-            if module_idx != function_ref.module_index || func_idx != function_ref.function_index {
-                // TODO: should we return a different error here?
+            if module_idx != function_ref.module_index {
```

Comment:
Maybe just one `if` statement here with both conditions combined?

Reply:
The reason I wanted to split them off was to be able to discern which reference is wrong - the module or the function. This was done to conform to the two errors: ModuleNotFound and FunctionNotFound. However, it would also be valid to say there should be a 3rd error type to indicate a stale or incorrect reference

Reply:
I see, that makes sense. I also agree on the 3rd error type for an incorrect reference.